### ISSUE TITLE
OPML works from file, not from URL

### ISSUE THREAD

**Issue by serussell** [newsboat/newsboat#410]
## Version

```
newsboat 2.14.0
System: Linux 4.20.0-arch1-1-ARCH (x86_64)
Compiler: g++ 8.2.1 20181127
ncurses: ncurses 6.1.20180127 (compiled with 6.1)
libcurl: libcurl/7.63.0 OpenSSL/1.1.1a zlib/1.2.11 libidn2/2.0.5 libpsl/0.20.2 (+libidn2/2.0.4) libssh2/1.8.0 nghttp2/1.35.1 (compiled with 7.63.0)
SQLite: 3.26.0 (compiled with 3.26.0)
libxml2: compiled with 2.9.8
```

## Summary

I have an OPML which, if downloaded with `curl` and imported with `-i`, works fine.  If the URL is specified in the config file with `urls-source` and `opml-url`, newsboat complains about no subscribed feeds.  I would expect the behavior to be the same, or if not, then the differences to be documented.

### Success case
1. `curl -O http://save.vs.totalpartykill.ca/grab-bag/osr.opml`
2. `newsboat -i osr.opml`
3. Result: successfully adds 277 feeds to `~/.newsboat/urls`:
```
➜  ~ wc -l .newsboat/urls
3 .newsboat/urls
➜  ~ newsboat -i osr.opml
Import of osr.opml finished.
➜  ~ wc -l .newsboat/urls
280 .newsboat/urls
```

### Failure case:
1. Add the following lines to `~/.newsboat/config`:
```
urls-source "opml"
opml-url "http://save.vs.totalpartykill.ca/grab-bag/osr.opml"
```
2. Run `newsboat`
3. Result:
```
Starting newsboat 2.14.0...
Loading configuration...done.
Opening cache...done.
Loading URLs from http://save.vs.totalpartykill.ca/grab-bag/osr.opml...done.
It looks like the OPML feed you subscribed contains no feeds. Please fill it with feeds, and try again.
```

## Additional

If the first case succeeds, I'd expect the second case to succeed. If using `urls-source` requires different XML tags or structure, it'd be great if that were documented.

**Comment by zaowen** [comment]
Thanks for the report! 
Reproduced in 66838f1f

**Comment by zaowen** [comment]
The function that imports opml files exists in 2 places. They should be consolidated.

**Comment by Minoru** [comment]
I don't think we should close this unless we change the error message. It's technically correct, but quite useless to the user. I'd count `<outline>`s that don't have `type="rss"`, and mention their number in the message like this: "It looks like the OPML file you subscribed to contains no feeds (123 outlines were ignored because they don't have type='rss' attribute). Please fill it with some feeds, and try again".

@zaowen, a PR with refactorings would be welcome!

**Comment by serussell** [comment]
Forgive my ignorance of the OPML specification.

> This is because the network opml file could be used for multiple things, not just rss readers 

Does this mean that an OPML can reasonably be expected to contain both feeds _and_ non-feeds?  I ask to see whether there's practical value to choosing the pedantic parser over the more _ahem_ liberal parser.  If it's going to commonly cause incorrect results because it's common to find OPMLs with mixed content, then I bow to strictness.

While my opinion doesn't have much value here, better error messages are always valuable.

Thanks to @zaowen for the detailed analysis!

**Comment by zaowen** [comment]
@serussell I'm honestly unsure if anyone does use an OPML for more than one list, I'm just afraid that making the change will break someone elses experience. If @Minoru think it would be alright to just ignore the `type="rss"` property, I'll gladly do it. It would even make the refactoring easier.

**Comment by Minoru** [comment]
I don't know how common mixed-content OPML is, and worse yet, I'm not sure how to estimate that. Googling for a "list of OPML files" doesn't give us an easy way out :) We could hand-crawl the Internet, compile our own list, and check that, but it's questionable how representative that sample would be.

I think @der-lyse is right and we will have to add an option to go with `opml-url`, with a default that preserves current behaviour. Here's how I would approach the implementation.
1. during import, count all `<outline>`s that have `xmlUrl` attribute but lack `type="rss"`. At the end, if the count is non-zero, tell the user that "N entries that look like feed URLs were ignored because their `type` attribute is not `rss`". This might be a warning separate from the error about "no feeds".
2. add an option that will relax the parser, letting it import anything with `xmlUrl` attribute. Add the option to the docs.
3. mention the option in the error message by adding something like "If you want to import those feeds, too, set `this and that` option to `true`" (or maybe `false`; that'll depend on the name of the option).

I'd make a separate commit for each of these points.

**Comment by serussell** [comment]
As a user, I appreciate this approach, as the alternative would prevent me from accomplishing something, or (more likely) force me to program some sort of proxy rewriter for the feed.  I recognize it smells hacky, but ... thanks.

**Comment by Minoru** [comment]
It's been a while since I last talked to Zaowen. Un-assigning since it's unlikely that they're working on this at the moment.
